src/lib.rs

```diff
@@ -15,15 +15,6 @@
 //! i.e. a [Sender]/[Receiver] pair and send and receive textual or
 //! binary data.
 //!
-//! **Note**: While it is possible to only receive websocket messages it is
-//! not possible to only send websocket messages. Receiving data is required
-//! in order to react to control frames such as PING or CLOSE. While those will be
-//! answered transparently they have to be received in the first place, so
-//! calling [`connection::Receiver::receive`] is imperative.
-//!
-//! **Note**: None of the `async` methods are safe to cancel so their `Future`s
-//! must not be dropped unless they return `Poll::Ready`.
```

Comment:
The doc comment on `SendToken` states that tokens are used to ensure that previous send operations are successful.

```rust
/// A `SendToken` is required in order to send websocket data.
///
/// The token serves as proof that the previous send operation was
/// completed successfully.
```

It does not state why previous send operations have to be successful in order for the next send operation to be legal.

```rust
//! **Note**: None of the `async` methods are safe to cancel so their `Future`s
//! must not be dropped unless they return `Poll::Ready`.
```

I find this note to be a great addition. Do you think it is worth adding either on the `*Token` doc comment or on the `send` / `receive` methods?

Reply:
I thought I could remove it because now the type-system ensures that futures are driven to completion but some such explanation could be added.